### PR TITLE
Feat: rework vela up to support specified revision

### PR DESCRIPTION
**Pull request by Somefive**
Signed-off-by: Somefive <yd219913@alibaba-inc.com>

### Description of your changes

Support `vela up --revision` to update application with a previous existing revision.

The rework support auto-completion for vela up.

Demo:
ezgif com-gif-maker

I have:

- [x] Read and followed KubeVela's [contribution process](https://github.com/oam-dev/kubevela/blob/master/contribute/create-pull-request.md).
- [ ] [Related Docs](https://github.com/oam-dev/kubevela.io) updated properly. In a new feature or configuration option, an update to the documentation is necessary. 
- [x] Run `make reviewable` to ensure this PR is ready for review.
- [ ] Added `backport release-x.y` labels to auto-backport this PR if necessary.

### CODE REVIEW

File: pkg/cmd/completion.go [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+
+	"github.com/spf13/cobra"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	cmdutil "k8s.io/kubectl/pkg/cmd/util"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	"github.com/oam-dev/kubevela/apis/core.oam.dev/v1beta1"
+	"github.com/oam-dev/kubevela/pkg/oam"
```
[the remark this answers is not in the document]
Comment (Somefive) [reply pkg/cmd/completion.go:30]: @sonatype-lift ignore

[the remark this answers is not in the document]
Comment (Somefive) [reply pkg/cmd/completion.go:30]: This is an invalid report.

File: pkg/cmd/builder.go [hunk 2]
```diff
@@ -0,0 +1,116 @@
+/*
+Copyright 2021 The KubeVela Authors.
```
Comment (wonderflow) [inline pkg/cmd/builder.go:2]:
```suggestion
Copyright 2021 - 2022 The KubeVela Authors.
```

**Review by wonderflow — APPROVED** [review]
great job

File: pkg/cmd/builder.go (hunk 2 above)
Comment (Somefive) [reply pkg/cmd/builder.go:2]: Done

**Review by barnettZQG — APPROVED** [review]
LGTM

**Review by leejanee — APPROVED** [review]

**Review by zzxwill — APPROVED** [review]

**Pull request merged**